plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/LanguageServerFileTypeRegister.java

```diff
@@ -90,40 +84,43 @@ public void start(final Callback<WsAgentComponent, Exception> callback) {
         registeredLanguages.then(new Operation<List<LanguageDescription>>() {
             @Override
             public void apply(List<LanguageDescription> langs) throws OperationException {
+                LOGGER.info("registering language descriptions");
                 if (!langs.isEmpty()) {
                     JsArrayString contentTypes = JsArrayString.createArray().cast();
                     for (LanguageDescription lang : langs) {
-                        String primaryExtension = lang.getFileExtensions().get(0);
-                        for (String ext : lang.getFileExtensions()) {
-                            final FileType fileType = new FileType(resources.file(), ext);
-                            fileTypeRegistry.registerFileType(fileType);
-                            editorRegistry.registerDefaultEditor(fileType, editorProvider);
-                            ext2langId.put(ext, lang.getLanguageId());
+                        if (lang.getFileExtensions() != null) {
```

Comment:
@tsmaeder yes LanguageDescription has a setter.
But the setter could modify the internal list only if the given element is not null instead of setting the given value directly without any check.

for example in LanguageDescription we could have
constructor : initialize fileExtensions to an empty list

in the setter: update this internal list only if the provided list is not null. (we could also make it immutable list by copying the elements of the provided list to the internal list and not reusing the provided list but it's another topic. Getter returning a copy of the list as well)

we could also in getter, return Collections.emptyList if the internal list is null. (either change setter logic or getter logic)

so, even with a setter, getFileExtensions is always returning a not null list